### PR TITLE
Add functionality to update profile and change password

### PR DESCRIPTION
**Pull request by newmskywalker**

### CODE REVIEW

File: shared/src/commonMain/kotlin/io.newm.shared/internal/repositories/UserRepository.kt [hunk 1]
```diff
@@ -39,30 +39,33 @@ internal class UserRepository(
...
-                        biography = user.biography,
-                        walletAddress = user.walletAddress,
-                        email = user.email,
-                        companyName = user.companyName,
-                        companyLogoUrl = user.companyLogoUrl,
-                        companyIpRights = user.companyIpRights,
-                        verificationStatus = user.verificationStatus,
-                    )
+                    dbWrapper().transaction {
+                        dbWrapper().userQueries.deleteAll() // just in case...
```
Comment (cristhianescobar) [inline shared/src/commonMain/kotlin/io.newm.shared/internal/repositories/UserRepository.kt:43]: 👍🏼

File: android/app-newm/src/main/java/io/newm/screens/profile/edit/ProfileEditUi.kt [hunk 2]
```diff
@@ -148,11 +161,25 @@ private fun ProfileScreenPreview() {
...
+                newPasswordState = TextFieldState(),
+                confirmPasswordState = TextFieldState(),
+                errorMessage = null,
                 eventSink = {},
             )
         )
     }
 }
+
+private fun User.toProfile(): Content.Profile {
```
Comment (cristhianescobar) [inline android/app-newm/src/main/java/io/newm/screens/profile/edit/ProfileEditUi.kt:178]: Does it make sense to have both user and profile models?

File: android/app-newm/src/main/java/io/newm/screens/profile/ProfileForm.kt [hunk 3]
```diff
@@ -68,26 +76,26 @@ fun ProfileForm(
                     style = formTitleStyle
                 )
                 Spacer(modifier = Modifier.height(16.dp))
-                TextFieldWithLabel(
-                    labelResId = R.string.profile_form_password_current_password,
-                    onValueChange = { value -> onProfileUpdated(profile.copy(currentPassword = value)) },
-                    isPassword = true,
-                    textfieldBackgroundColor = Gray23,
-                    placeholderResId = R.string.profile_placeholder_your_password
+                Password(
```
Comment (cristhianescobar) [inline android/app-newm/src/main/java/io/newm/screens/profile/ProfileForm.kt:79]: Do we need an email code to be able to update password? I assume we don't just double-checking

**Review by cristhianescobar — APPROVED** [review]
LGTM

File: android/app-newm/src/main/java/io/newm/screens/profile/edit/ProfileEditUi.kt (hunk 2 above)
Comment (newmskywalker) [reply android/app-newm/src/main/java/io/newm/screens/profile/edit/ProfileEditUi.kt:178]: I think it provides better separation of concerns. 
If we need to update the User class for whatever reason (or even delete it), we won't have to refactor the UI as well

File: android/app-newm/src/main/java/io/newm/screens/profile/ProfileForm.kt (hunk 3 above)
Comment (newmskywalker) [reply android/app-newm/src/main/java/io/newm/screens/profile/ProfileForm.kt:79]: We don't, it's only needed to update the email

**Pull request merged**